sdks/python/apache_beam/runners/portability/fn_api_runner_transforms.py

```diff
@@ -142,6 +142,135 @@ def leaf_transform_stages(
+              spec=beam_runner_api_pb2.FunctionSpec(
+                  urn=common_urns.coders.KV.urn)),
+          component_coder_ids=[key_coder_id, accumulator_coder_id])
+      key_accumulator_coder_id = context.add_or_get_coder_id(
+          key_accumulator_coder)
+
+      accumulator_iter_coder = beam_runner_api_pb2.Coder(
+          spec=beam_runner_api_pb2.SdkFunctionSpec(
+              spec=beam_runner_api_pb2.FunctionSpec(
+                  urn=common_urns.coders.ITERABLE.urn)),
+          component_coder_ids=[accumulator_coder_id])
+      accumulator_iter_coder_id = context.add_or_get_coder_id(
+          accumulator_iter_coder)
+
+      key_accumulator_iter_coder = beam_runner_api_pb2.Coder(
+          spec=beam_runner_api_pb2.SdkFunctionSpec(
+              spec=beam_runner_api_pb2.FunctionSpec(
+                  urn=common_urns.coders.KV.urn)),
+          component_coder_ids=[key_coder_id, accumulator_iter_coder_id])
+      key_accumulator_iter_coder_id = context.add_or_get_coder_id(
+          key_accumulator_iter_coder)
+
+      precombined_pcoll_id = unique_name(
+          context.components.pcollections, 'pcollection')
+      context.components.pcollections[precombined_pcoll_id].CopyFrom(
+          beam_runner_api_pb2.PCollection(
+              unique_name=transform.unique_name + '/Precombine.out',
+              coder_id=key_accumulator_coder_id,
+              windowing_strategy_id=input_pcoll.windowing_strategy_id,
+              is_bounded=input_pcoll.is_bounded))
+
+      grouped_pcoll_id = unique_name(
+          context.components.pcollections, 'pcollection')
+      context.components.pcollections[grouped_pcoll_id].CopyFrom(
+          beam_runner_api_pb2.PCollection(
+              unique_name=transform.unique_name + '/Group.out',
+              coder_id=key_accumulator_iter_coder_id,
+              windowing_strategy_id=output_pcoll.windowing_strategy_id,
+              is_bounded=output_pcoll.is_bounded))
+
+      merged_pcoll_id = unique_name(
+          context.components.pcollections, 'pcollection')
+      context.components.pcollections[merged_pcoll_id].CopyFrom(
+          beam_runner_api_pb2.PCollection(
+              unique_name=transform.unique_name + '/Merge.out',
+              coder_id=key_accumulator_coder_id,
+              windowing_strategy_id=output_pcoll.windowing_strategy_id,
+              is_bounded=output_pcoll.is_bounded))
+
+      def make_stage(base_stage, transform):
+        return Stage(
+            transform.unique_name,
+            [transform],
+            downstream_side_inputs=base_stage.downstream_side_inputs,
+            must_follow=base_stage.must_follow)
+
+      yield make_stage(
+          stage,
+          beam_runner_api_pb2.PTransform(
+              unique_name=transform.unique_name + '/Precombine',
+              spec=beam_runner_api_pb2.FunctionSpec(
+                  urn=common_urns.combine_components.COMBINE_PGBKCV.urn,
```

Comment:
We do we use this Combine URN and not the other ones here: https://github.com/apache/beam/blob/c2a9cac248ec45fbbd35bacec7de26321fc8d69f/model/pipeline/src/main/proto/beam_runner_api.proto#L252 ?

Could we unify them? 

What does PGBKCV stand for? :)

Reply:
Yeah, the duplication of combiner URNs was quite a surprise to me too (and took a while to debug--I kept checking that we're pulling the Java and Python constants from the (freshly generated) proto files and couldn't figure out why the Java registrations weren't getting picked up by the Python-generated transforms until I actually printed the values out--so much for once source of truth :-)...) I filed BEAM-6199 to resolve this; the comment at https://github.com/apache/beam/pull/5128/commits/030749ffa4ee00db0a6fe39cc308b330df844147 seems to indicate there are still references lying around elsewhere.

Also, PGBKCV = PartialGroupByKeyCombiningValues.